changefeedccl: return verbose error upon webhook sink context cancellation

spiffyy99 opened the pull request:
Previously, when the webhook sink workers were canceled (via context),
the sink returned a `context canceled` error. This change adds a check
to the inflight tracker for the HTTP error which will be clearer to
the user.

Release note: None

Specifically, this check is done within `EmitRow()`. We do this check at the top of the function, but there is a possibility that context cancellation will occur between the initial check and the batch channel send, so we re-check the error to make sure the user sees a proper reason for the changefeed failure, as opposed to `context canceled`.

Also added a test for the failure where an HTTP server is used (vs HTTPS) which was a failure a user experienced.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69160)
<!-- Reviewable:end -->

spiffyy99 committed:
changefeedccl: return verbose error upon webhook sink context cancellation

Previously, when the webhook sink workers were canceled (via context),
the sink returned a `context canceled` error. This change adds a check
to the inflight tracker for the HTTP error which will be clearer to
the user.

Release note: None

spiffyyeng commented:
closing as this has become part of a bigger issue (will open new PR)
